google-api-client/src/main/java/com/google/api/client/googleapis/batch/BatchRequest.java

```diff
@@ -213,6 +230,13 @@ public int size() {
   public void execute() throws IOException {
     boolean retryAllowed;
     Preconditions.checkState(!requestInfos.isEmpty());
+
+    // Log a warning if the user is using the global batch endpoint. In the future, we can turn this
+    // into a preconditions check.
+    if (this.batchUrl.toString().equals(GLOBAL_BATCH_ENDPOINT)) {
```

Comment:
I would flip the equality here to avoid a NPE.

Reply:
Done.